qiita_db/study.py

```diff
@@ -673,7 +674,13 @@ def sample_template(self):
         -------
         SampleTemplate id
         """
-        return self._id
+        with TRN:
+            sql = """SELECT EXISTS(SELECT *
+                                   FROM qiita.study_sample
+                                   WHERE study_id = %s)"""
+            TRN.add(sql, [self.id])
+            exists = TRN.execute_fetchlast()
+        return self._id if exists else None
```

Comment:
I don't generally like the idea of having different return types for a property, not sure how others feel? Would you mind updating the docstring to indicate that the returned value is an int and that it can be None?